engine/runtime/src/main/java/org/enso/compiler/TreeToIr.java

```diff
@@ -690,6 +702,38 @@ yield switch (op.codeRepr()) {
       case Tree.Group group -> {
         yield translateExpression(group.getBody(), moreArgs, insideTypeSignature, isMethod);
       }
+      case Tree.TextLiteral txt -> {
+        var fullTxt = txt.codeRepr().trim();
+        var t = fullTxt.substring(1, fullTxt.length() - 1);
```

Comment:
So it is ignoring `"`?

Reply:
Yes. I'd like this to be handled by the parser itself, but that is not the case yet. I'd like the full text literal to be provided to me by the parser, but @kazcw explained to me that's not possible - I can only get it line by line. Still quoting, character conversions, etc. would better be handled by the parser itself.

Reply:
> I'd like the full text literal to be provided to me by the parser, but @kazcw explained to me that's not possible

Right, the problem is not well-defined. A `TextLiteral` that can contain splices cannot in general be represented as a `String`. I think splices are new (it looks like they aren't yet representable in the IR?), so from the perspective of implementing `TreeToIr` right now this representation might look overly-complicated; but if the parser provided an interface producing a `String` that only worked when the input didn't contain splices, we'd have no use for it once we had a general implementation that handles any `TextLiteral` as a sequence of text sections and expressions. Your labor tying strings together now is the foundation of a new language feature.

> Still quoting, character conversions, etc. would better be handled by the parser itself.

They are. `codeRepr` is not the parsed text, it stringifies back to the literal source code... The parser handles escapes, splices, quotes, and the block text left-trim algorithm, and stores the result in the `TextLiteral`'s fields. Call `getElements` and iterate to use the *parsed* data.

Reply:
OK, I got it. I am supposed to use `getElements`...

> hold off on implementing that until https://github.com/enso-org/enso/pull/3716 is merged though

...once #3716 is integrated. Meanwhile keeping this `fullTxt.substring(1, fullTxt.length() - 1)` hack is enough.